controller/node_controller.go

```diff
@@ -636,15 +656,24 @@ func (nc *NodeController) syncDiskStatus(node *longhorn.Node) error {
 			} else {
 				diskUUID = diskConfig.DiskUUID
 			}
-			if diskStatusMap[id].DiskUUID == "" {
-				if len(disks) > 1 {
-					diskStatusMap[id].Conditions = types.SetConditionAndRecord(diskStatusMap[id].Conditions,
-						types.DiskConditionTypeReady, types.ConditionStatusFalse,
+
+			if nc.checkSameFsidInDisks(id, disks,
+				diskStatusMap) {
+				// Found multiple disks in the same Fsid
+				diskStatusMap[id].Conditions =
+					types.SetConditionAndRecord(
+						diskStatusMap[id].Conditions,
+						types.DiskConditionTypeReady,
+						types.ConditionStatusFalse,
 						string(types.DiskConditionReasonDiskFilesystemChanged),
 						fmt.Sprintf("Disk %v(%v) on node %v is not ready: disk has same file system ID %v as other disks %+v", id, disk.Path, node.Name, fsid, disks),
-						nc.eventRecorder, node, v1.EventTypeWarning)
-					continue
-				}
+						nc.eventRecorder, node,
+						v1.EventTypeWarning)
+				continue
+
+			}
+
+			if diskStatusMap[id].DiskUUID == "" {
```

Comment:
Why move this line? If the disk already has UUID assigned, we can use logic on line 691 to validate the disk.

Reply:
@yasker sorry, I don't understand this comment, I moved the `if len(disks) > 1 {` out of `if diskStatusMap[id].DiskUUID == "" {`.

Reply:
I meant, why you moved it? Previously we only check FSID if the UUID doesn't exist yet for the disk. Now we will apply FSID on every disk as the result of moving it, right?